src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.Simple.cs

```diff
@@ -21,11 +40,15 @@ public partial class CborWriter
         /// </exception>
         public void WriteSingle(float value)
         {
-            EnsureWriteCapacity(5);
-            WriteInitialByte(new CborInitialByte(CborMajorType.Simple, CborAdditionalInfo.Additional32BitData));
-            BinaryPrimitives.WriteSingleBigEndian(_buffer.AsSpan(_offset), value);
-            _offset += 4;
-            AdvanceDataItemCounters();
+            if (!CborConformanceModeHelpers.RequiresPreservingFloatPrecision(ConformanceMode) &&
+                 FloatSerializationHelpers.TryConvertSingleToHalf(value, out Half half))
+            {
+                WriteHalf(half);
```

Comment:
Why `WriteHalf` vs `WriteSingleCore` and `WriteDoubleCore`?

Reply:
The `Write[Float]Core` methods implement serialization for the particular precision without doing conformance mode checks and potential downsizing. Perhaps calling it something like `WriteSingleNoDownsizing` might reduce confusion?